src/envoy/http/jwt_auth/jwt_authenticator.cc

```diff
@@ -195,8 +195,12 @@ void JwtAuthenticator::VerifyKey(const PubkeyCacheItem& issuer_item) {
     return;
   }
 
-  // TODO: check forward_payload_header.
   headers_->addReferenceKey(kJwtPayloadKey, jwt_->PayloadStrBase64Url());
+  const LowerCaseString key(issuer_item.jwt_config().forward_payload_header());
+  if (!key.get().empty() && !jwt_->PayloadStrBase64Url().empty() &&
```

Comment:
not need to check jwt_->PayloadStrBaseUrl()  a valid JWT will not have an empty Payload

I felt it is better to check forward_payload_header first, then create LowerCaseString.

Reply:
Done.